torchtune/modules/transformer.py

```diff
@@ -52,6 +90,10 @@ def forward(
                 and column j means token i attends to token j. A value of False means token i
                 does not attend to token j. If no mask is specified, a causal mask
                 is used by default. Default is None.
+            encoder_input (Optional[Tensor]): input to keep a consistent signature accross
+                transformer layers. This input is ignored and does nothing in self attention.
```

Comment:
you could fold these into kwargs in the TransformerLayer protocol, then you don't have to define these for the self attention layer. This is awkward otherwise

Reply:
Why not just add **kwargs to the forward here?